packages/realm-react/src/__tests__/useEmailPasswordAuth.test.tsx

```diff
@@ -0,0 +1,241 @@
+      });
+
+      // Get an instance of the realm app and make sure the current user has been set
+      const realmApp = new App({ id: appId, baseUrl });
+      expect(realmApp.currentUser).not.toBeNull();
+      await act(async () => {
+        await result.current.logOut();
+      });
+      expect(realmApp.currentUser).toBeNull();
+    });
+    it("can register and login with email/password, just by calling register and login", async () => {
+      const { result } = renderEmailPasswordAuth(appId, baseUrl);
+      await act(async () => {
+        result.current.register({ email: "test2@test.com", password: "password", loginAfterRegister: false });
+        await waitFor(() => {
+          expect(result.current.result.pending).toEqual(true);
+        });
+      });
+      await waitFor(() => {
+        expect(result.current.result.success).toEqual(true);
+      });
+
+      await act(async () => {
+        result.current.logIn({ email: "test2@test.com", password: "password" });
+        await waitFor(() => {
+          expect(result.current.result.pending).toEqual(true);
+        });
+      });
+      await waitFor(() => {
+        expect(result.current.result.success).toEqual(true);
+      });
+
+      // Get an instance of the realm app and make sure the current user has been set
+      const realmApp = new App({ id: appId, baseUrl });
+      const user = realmApp.currentUser;
+      expect(user).not.toBeNull();
+
+      await act(async () => {
+        await result.current.logOut();
+      });
+      expect(realmApp.currentUser).toBeNull();
+    });
+    it("sets an error state when user is already registered", async () => {
+      const { result } = renderEmailPasswordAuth(appId, baseUrl);
+
+      await act(async () => {
+        result.current.register({ email: "test2@test.com", password: "password", loginAfterRegister: false }),
+          await waitFor(() => {
+            expect(result.current.result.pending).toEqual(true);
+          });
+      });
+      await waitFor(() => {
+        expect(result.current.result.state).toBe(OperationState.Error);
+        expect(result.current.result.error).toBeDefined();
+      });
+    });
+  });
+  describe("all methods are callable and report a state", () => {
+    let appId: string;
+    beforeAll(async () => {
+      const config = new AppConfigBuilder("test-app-2");
+      config.authProvider({
+        name: "local-userpass",
+        type: "local-userpass",
+        config: {
+          autoConfirm: true,
+          resetPasswordUrl: "http://localhost/resetPassword",
+        },
+        disabled: false,
+      });
+      ({ appId } = await importApp(config.config));
+    });
+    it("logIn", async () => {
+      const { result } = renderEmailPasswordAuth(appId, baseUrl);
+      await act(async () => {
+        result.current.logIn({ email: "test@test.com", password: "password" });
+        await waitFor(() => {
+          expect(result.current.result.pending).toBeTruthy();
```

Comment:
Why not `toEqual(true)`?

Reply:
On one point they should be consistent.  `toBeTruthy` would guarantee that the value resolves to true in an `if` statement.